src/disasm.rs

```diff
@@ -0,0 +1,159 @@
+
+                // Process the first line
+                match lines_iter.next() {
+                    Some(line) => match line.split_once(':') {
+                        Some((file_name, leftover_line)) => {
+                            disasm.update_name(file_name);
+                            match leftover_line.trim().strip_prefix("file format ") {
+                                Some(file_format) => {
+                                    disasm.update_format(file_format);
+                                }
+                                None => {
+                                    return Err("Incorrect format for the first line".to_string());
+                                }
+                            }
+                        }
+                        None => {
+                            return Err("Incorrect format for the first line".to_string());
+                        }
+                    },
+                    None => {
+                        return Err("Error, the file does not contain any text".to_string());
+                    }
+                }
+
+                // Process all other lines
+                for line in lines_iter {
+                    let line = line.trim();
+                    if let Some(section) = line.strip_prefix("Disassembly of section ") {
+                        match section.trim().strip_suffix(":") {
+                            Some(section_name) => {
+                                disasm.add_section(Section::new(section_name));
+                            }
+                            None => {
+                                return Err("Error parsing one section name".to_string());
+                            }
+                        }
+                    } else if line.starts_with("<") {
+                        match line.trim().strip_suffix(":") {
+                            Some(symbol_name) => {
+                                if let Err(msg) = disasm.add_symbol(Symbol::new(symbol_name)) {
+                                    return Err(msg);
+                                }
+                            }
+                            None => {
+                                return Err("Error parsing one section name".to_string());
+                            }
+                        }
+                    } else {
+                        if let Err(msg) = disasm.add_instruction(Instruction::from(line)) {
+                            return Err(msg);
+                        }
+                    }
+                }
+                Ok(disasm)
+            }
+            Err(msg) => Err(format!(
+                "Error reading a line of the disassembly file :{}",
+                msg
+            )),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn empty_section_with_no_name() {
+        let disasm = Disasm::try_from(BufReader::new(File::open("test-output").unwrap()));
```

Comment:
It is unclear to me what would be a good way to mock the `File::open()` and `BufReader::new()` functions in these unit tests. 
Any suggestion?